blocks/initialization.py

```diff
@@ -147,21 +148,39 @@ def __init__(self, mult=1):
         self.mult = mult
 
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
-        rows, cols = shape
-        return self.mult * numpy.eye(rows, cols, dtype=theano.config.floatX)
+        if len(shape) == 1:
+            eye_matrix = numpy.eye(1, shape[0], dtype=theano.config.floatX)
+            return self.mult * eye_matrix[0]
+        elif len(shape) == 2:
+            rows, cols = shape
+            eye_matrix = numpy.eye(rows, cols, dtype=theano.config.floatX)
+            return self.mult * eye_matrix
+        else:
+            raise ValueError("Identity initialization only works for 1D and"
+                             " 2D variables")
 
 
 class Orthogonal(NdarrayInitialization):
-    """Initialize a random orthogonal matrix.
+    r"""Initialize a random, approximately orthogonal matrix.
+
+    Only works for 1D and 2D variables.
 
-    Only works for 2D arrays.
+    When used to initialize a 1D parameter vector the elements will
+    be drawn i.i.d. from a normal distribution.
+
+    For 2D matrices we initialalize such that
+    :math:`M M^T \approx I` and :math:`M^T M \approx I` where
+    :math:`I` is an identity matrix of appropriate size.
 
     """
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
+        if len(shape) not in (1, 2):
+            raise ValueError("Orthogonal initialization only works for 1D and"
+                             " 2D variables")
+
+        if len(shape) == 1:
+            # Special case for 1D variables
+            return rng.randn(*shape).astype(theano.config.floatX)
```

Comment:
> So what?

If the name has no relation to what is output, I suggest `Uniform` should from now on output a Weibull distribution on Thursdays, but only if the weather is nice.

If you want to special case things for LSTM, I don't see the problem of writing:
```python
OrthogonalLSTM(NdarrayInitialization):
    """Orthogonal initialization on weight matrices, random initialization for the gates."""
    def generate(self, rng, shape):
        if len(shape) == 1:
            return IsotropicGaussian().generate(rng, shape)
        if len(shape) == 2:
            return Orthogonal().generate(rng, shape)
```